apps/application-system/api/src/app/modules/application/template-api.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable, NotFoundException } from '@nestjs/common'
+import { ApplicationService } from './application.service'
+
+import { BaseTemplateApiApplicationService } from '@island.is/application/template-api-modules'
+
+@Injectable()
+export class TemplateApiApplicationService extends BaseTemplateApiApplicationService {
+  constructor(private readonly applicationService: ApplicationService) {
+    super()
+  }
+
+  async saveAttachmentToApplicaton(
+    applicationId: string,
+    key: string,
+    url: string,
+  ): Promise<void> {
+    const existingApplication = await this.applicationService.findOneById(
+      applicationId,
+    )
+
+    if (!existingApplication) {
+      throw new NotFoundException(
+        `An application with the id ${applicationId} does not exist`,
+      )
+    }
```

Comment:
Should accept an application object. No need to check if it exists.

The `key` and `url` arguments feel too low level since we don't want anyone to mess with either. Maybe we should move the upload and key logic here? Could also reuse logic from S3/FileService.